### ISSUE TITLE
Mozrepl not working in standalone

### ISSUE THREAD

**Issue by averter** [vspinu/zotelo#34]
Reading this page of the wiki https://github.com/vspinu/zotelo/wiki/Standalone
I got to this point where it says  "Once you do, the "Tools" menu will have a new entry named "MozRepl". but unfortunately I don't see it. 
I am using Zotero 5.0.29 and Mozrepl Version 1.1.2.1-signed.1-signed. I am not sure if this is the latest version from Mozrepl, since in the main page I read a warning "This add-on is not compatible with your version of Firefox." I have: restarted zotero and; tried to run zotelo from emacs still without luck. Thanks in advance for any help.

**Comment by a-fent** [comment]
Firefox 57.0 ("Quantum") has many big changes which mean that MozRepl will no longer be supported. https://github.com/bard/mozrepl

As an alternative to zotelo, I would suggest looking at emacs-zotero-bib-fetch.el https://gist.github.com/andersjohansson/fa7ca643782771b6e15da41514e1358a

It borrows the code and idea from zotelo, but uses the Zotero BetterBibtex extension. I can confirm that this set up works well, including with the new version of Zotero (5.0), for which the BetterBibtex extension (you have to install this) has been updated.

To @vspinu : would you be open to simply merging the Better Bibtex-based implementation, since the MozRepl approach is essentially dead?

**Comment by vspinu** [comment]
> To @vspinu : would you be open to simply merging the Better Bibtex-based implementation, since the MozRepl approach is essentially dead?

Yes, it would be great. I have just switched to zotero standalone myself as firefox 57 no longer supports zotero, so no way to evade it further :) I will try to have a look. As always, any patches with   minimal changes to the UI are very much welcome.

**Comment by averter** [comment]
Thanks @a-fent. That suits me quite well, since I've been using the better bibtex for a long time as well. 
Shamefully I am not an elisp expert myself, and I cannot put the ezbf minor mode to work. Specifically 

- I added `(add-hook 'TeX-mode-hook 'ezbf-minor-mode)` to my .emacs file; 
- placed the el file in a directory where I have other el files "/usr/local/share/emacs/site-lisp/"; 
- and this folder was already added as a load path in .emacs with `(add-to-list 'load-path "/usr/local/share/emacs/site-lisp/")` 
Unfortunately it does not work, and conflicts with other minor-modes such as reftex, which stop working. Any ideas why this might be happening? Thanks

**Comment by a-fent** [comment]
@averter when you say "does not work", what does not work? Your set-up sounds fine with `add-hook …` etc. Is there anything appearing in the *Messages* buffer that might indicate what's going wrong? 

The interface should be pretty much the same as for zotelo, i.e. you press `C-c z u`, and if no collection has been set for the current document, it will prompt for one, in the shape of a Better Bibtex collection URL. If one is set, the bibliography wil be updated.

@vspinu ok, thanks, I'll try and prepare a patch in the next week or so. The UI is virtually the same, I think, just the implementation needs to be swapped in and the documentation updated.

**Comment by averter** [comment]
@a-fent I try to invoke those keybindings as in zotelo, and emacs says that the combination is undefined. Moreover, other minor-modes such as reftex also stop working. I can paste my whole .emacs here, although it is quite lengthy. Maybe just the Latex sections would be enough?

**Comment by a-fent** [comment]
@averter if the key combinations are undefined, it is likely that the mode is not getting loaded correctly. If you want to check this, you could try just putting the file `emacs-zotero-bib-fetch.el` in your ~/.emacs.d/ directory and calling `(load-file "~/.emacs.d/emacs-zotero-bib-fetch.el")` in your startup.

**Comment by averter** [comment]
@a-fent Thanks. it is working now. I am a new user to zotelo so I didn't know exactly what to expect. I am not sure of this, but... zotelo seems to simply mimic the automatic export feature of better bibtex. My original objective was to have an emacs replacement to the zotero GUI, without having to use the mouse. Does zotelo (or ezbf) allow to control the contents of collections as well?

**Comment by a-fent** [comment]
So far as I know, there is nothing for Emacs that allows both sides (Zotero & Emacs) both to read and write to the same database. There's a few ways it could be done but they would be complex to implement. Here I think also slightly misunderstood zotelo's description of "synchronising" the databases when I first looked into it.

That said, if you want to write (latex) in Emacs using Zotero bibliography, there are still quite a few other packages that might be of interest. Take a look at my answer at: https://emacs.stackexchange.com/questions/24725/zotero-workflow-into-emacs/24727#24727

The built-in packages `reftex` and possibly `ebib` would be worth looking it, and I definitely recommend looking at `org-ref`, which, despite the name, is useful when writing latex directly as well as writing `org` documents.

**Comment by averter** [comment]
This is all very interesting. There are so many tools with overlapping functionality that chosing the best framework for such a complex task becomes difficult. I am interested in the helm-bibtex + reftex combo. If I can get helm-bibtex to mimic the better bibtex export/update bibtex/biblatex files I think I'll replace zotero. Anyway, I think the issue can be closed now. Thanks for all the information/help.
